src/equations/3d/compressible_euler.jl

```diff
@@ -621,6 +657,121 @@ function flux_hll(u_ll, u_rr, orientation, equation::CompressibleEulerEquations3
 end
 
 
+ """
+    flux_hllc(u_ll, u_rr, orientation, equation::CompressibleEulerEquations3D)
+
+    Computes the HLLC flux (HLL with Contact) for compressible Euler equations developed by E.F. Toro
+    http://www.prague-sum.com/download/2012/Toro_2-HLLC-RiemannSolver.pdf
+"""
+function flux_hllc(u_ll, u_rr, orientation, equation::CompressibleEulerEquations3D)
+  # Calculate primitive variables and speed of sound
+  rho_ll, rho_v1_ll, rho_v2_ll, rho_v3_ll, rho_e_ll = u_ll
+  rho_rr, rho_v1_rr, rho_v2_rr, rho_v3_rr, rho_e_rr = u_rr
+
+  v1_ll = rho_v1_ll / rho_ll
+  v2_ll = rho_v2_ll / rho_ll
+  v3_ll = rho_v3_ll / rho_ll
+  e_ll  = rho_e_ll / rho_ll
+  p_ll = (equation.gamma - 1) * (rho_e_ll - 1/2 * rho_ll * (v1_ll^2 + v2_ll^2 + v3_ll^2))
+  c_ll = sqrt(equation.gamma*p_ll/rho_ll)
+
+  v1_rr = rho_v1_rr / rho_rr
+  v2_rr = rho_v2_rr / rho_rr
+  v3_rr = rho_v3_rr / rho_rr
+  e_rr  = rho_e_rr / rho_rr
+  p_rr = (equation.gamma - 1) * (rho_e_rr - 1/2 * rho_rr * (v1_rr^2 + v2_rr^2 + v3_rr^2))
+  c_rr = sqrt(equation.gamma*p_rr/rho_rr)
+
+  # Obtain left and right fluxes
+  f_ll = calcflux(u_ll, orientation, equation)
+  f_rr = calcflux(u_rr, orientation, equation)
+
+  if orientation == 1 # x-direction
+    vel_L = v1_ll 
+    vel_R = v1_rr
+  elseif orientation == 2 # y-direction
+    vel_L = v2_ll
+    vel_R = v2_rr
+  else # z-direction
+    vel_L = v3_ll
+    vel_R = v3_rr
+  end
+  sMu_L = -c_ll
+  sMu_R =  c_rr
+  Ssl = vel_L + sMu_L
+  Ssr = vel_R + sMu_R
+  M_ll = abs(vel_L)/c_ll
+  M_rr = abs(vel_R)/c_rr
+
+  #if Ssl >= 0.0 && Ssr > 0.0
+  if Ssl >= 0.0 
+    f1 = f_ll[1]
+    f2 = f_ll[2]
+    f3 = f_ll[3]
+    f4 = f_ll[4]
+    f5 = f_ll[5]
+  #elseif Ssr <= 0.0 && Ssl < 0.0
+  elseif Ssr <= 0.0 
+    f1 = f_rr[1]
+    f2 = f_rr[2]
+    f3 = f_rr[3]
+    f4 = f_rr[4]
+    f5 = f_rr[5]
+  else
+    SStar = (p_rr - p_ll + rho_ll*vel_L*sMu_L - rho_rr*vel_R*sMu_R) / (rho_ll*sMu_L - rho_rr*sMu_R)
+    if Ssl <= 0.0 && SStar >= 0.0
```

Comment:
Is this correct that here only `Ssl` is checked and the value of `Ssr` is irrelevant? It might very well be ok, it just looked weird to me since it seems to prefer `Ssl` over `Ssr` somehow.

Reply:
I'll check again just to be sure.

Reply:
Seems correct, according to the source I put into the documentation.